canonical-web-and-design/maas-ui
# fix(ui): allow admin to change user passwords

## Done
* Correctly dispatch `user.admin_change_password` when editing a user from settings (thanks @sparkiegeek :D)
* Migrate `UserForm` to typescript
* Add html autocomplete attributes to UserForm username and password fields
* Update `User` type with missing properties

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- As an admin change a user's password from settings/user. 
- Confirm that the password has changed correctly by logging in as that user with the new password.

## Fixes

Fixes: #1615 

## Launchpad issue

lp#1894727

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
